text/0032-minimal-relay.md

```diff
@@ -0,0 +1,190 @@
+
+- Parachains that interact with affected logic on the Relay Chain;
+- Core protocol and XCM format developers;
+- Tooling, block explorer, and UI developers.
+
+## Explanation
+
+The following pallets and subsystems are good candidates to migrate from the Relay Chain:
+
+- Identity
+- Preimage
+- Balances
+- Staking
+	- Staking
+	- Election Provider
+	- Bags List
+	- NIS
+	- Nomination Pools
+	- Fast Unstake
+- Governance
+	- Treasury and Bounties
+	- Conviction Voting
+	- Referenda
+
+Note: The Auctions and Crowdloan pallets will be replaced by Coretime, its system chain and
+interface described in RFC-1 and RFC-5, respectively.
+
+### Migrations
+
+Some subsystems are simpler to move than others. For example, migrating Identity can be done by
+simply preventing state changes in the Relay Chain, using the Identity-related state as the genesis
+for a new chain, and launching that new chain with the genesis and logic (pallet) needed.
+
+Other subsystems cannot experience any downtime like this because they are essential to the
+network's functioning, like Staking and Governance. However, these can likely coexist with a
+similarly-permissioned system chain for some time, much like how "Gov1" and "OpenGov" coexisted at
+the latter's introduction.
+
+Specific migration plans will be included in release notes of runtimes from the Polkadot Fellowship
+when beginning the work of migrating a particular subsystem.
+
+### Interfaces
+
+The Relay Chain, in many cases, will still need to interact with these subsystems, especially
+Staking and Governance. These subsystems will require making some APIs available either via
+dispatchable calls accessible to XCM `Transact` or possibly XCM `Instruction`s in future versions.
+
+For example, Staking provides a pallet-API to register points (e.g. for block production) and
+offences (e.g. equivocation). With Staking in a system chain, that chain would need to allow the
+Relay Chain to update validator points periodically so that it can correctly calculate rewards.
+
+A pub-sub protocol may also lend itself to these types of interactions.
+
+### Deployment
+
+Actual migrations should happen based on some prioritization. This RFC proposes to migrate Identity,
+Staking, and Governance as the systems to work on first. A brief discussion on the factors involved
+in each one:
+
+#### Identity
+
+Identity will be one of the simpler pallets to migrate into a system chain, as its logic is largely
+self-contained and it does not "share" balances with other subsystems. As in, any DOT is held in
+reserve as a storage deposit and cannot be simultaneously used the way locked DOT can be locked for
+multiple purposes.
+
+Therefore, migration can take place as follows:
+
+1. A runtime upgrade will block all calls to the pallet, preventing updates to identity info.
+1. The frozen state will form the genesis of a new system parachain.
+1. A Relay Chain runtime upgrade will allow migrating the deposit to the parachain. The parachain
```

Comment:
how does one get refunds of the reserves?

Reply:
This is quite simple and I've already written a quick implementation.

1. Add a function to `reap_identity` that will:
    1. Unreserve the deposit;
    2. Teleport enough (about 1/100th) to the parachain;
    3. Call a `poke_deposit` function that will reserve the deposit on the parachain.